numba/targets/arraymath.py

```diff
@@ -517,6 +517,24 @@ def flat_any(a):
 
     return flat_any
 
+@overload(np.average)
+def np_average(arr, weights=None):
+    if weights is None or isinstance(weights, types.NoneType):
+        def np_average_impl(arr, weights=None):
+            return np.mean(arr)
+
+    else:
+        def np_average_impl(arr, weights=None):
+            weighted_arr = np.multiply(arr, weights)
+            c = 0
+            cw = 0
+            for v in np.nditer(weighted_arr):
+                c += v.item()
+            for w in np.nditer(weights):
+                cw += w.item()
+            return c/cw
```

Comment:
If `cw` is zero this should raise a `ZeroDivisionError`, perhaps check the NumPy behaviour?